geomstats/information_geometry/normal.py

```diff
@@ -1,29 +1,73 @@
-"""Statistical Manifold of normal distributions with the Fisher metric.
+"""Information Manifold of multivariate normal distributions with the Fisher metric.
 
-Lead author: Alice Le Brigant.
+Lead authors: Antoine Collas, Alice Le Brigant.
 """
 
-from scipy.stats import norm
+import math
 
+from scipy.stats import multivariate_normal, norm
+
+import geomstats
 import geomstats.backend as gs
-import geomstats.errors
+import geomstats.errors as errors
+from geomstats.geometry.base import OpenSet
+from geomstats.geometry.euclidean import Euclidean
 from geomstats.geometry.poincare_half_space import (
     PoincareHalfSpace,
     PoincareHalfSpaceMetric,
 )
+from geomstats.geometry.product_manifold import ProductManifold
+from geomstats.geometry.product_riemannian_metric import NFoldMetric
 from geomstats.geometry.pullback_metric import PullbackDiffeoMetric
+from geomstats.geometry.riemannian_metric import RiemannianMetric
+from geomstats.geometry.spd_matrices import SPDAffineMetric, SPDMatrices
 from geomstats.information_geometry.base import InformationManifoldMixin
 
 
-class NormalDistributions(InformationManifoldMixin, PoincareHalfSpace):
+class NormalDistributions:
+    """Class for the normal distributions.
+
+    This class is a common interface to the following different situations:
+
+    - univariate normal distributions
+    - centered multivariate normal distributions
+    - multivariate normal distributions with diagonal covariance matrix
+    - general multivariate normal distributions
+
+    Parameters
+    ----------
+    sample_dim : int
+        Dimension of the sample space of the normal distribution.
+    distribution_type : str, {'centered', 'diagonal', 'general'}
+        Type of distributions.
+        Optional, default: 'general'.
+    """
+
+    def __new__(cls, sample_dim, distribution_type="general"):
+        """Instantiate class that corresponds to the distribution_type."""
+        errors.check_parameter_accepted_values(
+            distribution_type,
+            "distribution_type",
+            ["centered", "diagonal", "general"],
+        )
+        if sample_dim == 1:
```

Comment:
what if "centered" and `sample_dim == 1`?

Reply:
If sample_dim is one, we don't need to distinguish between special cases since everything is explicit for univariate distributions.